src/symbolic.jl

```diff
@@ -84,22 +84,32 @@ end
 
 function SymbolicModel(data::Dict, filename="none")
     # verify that we have all the required fields
-    for k in (:symbols, :equations, :calibration, :exogenous)
+    for k in (:symbols, :equations, :calibration)
         if !haskey(data, k)
             error("Yaml file must define section $k for dtcc model")
         end
     end
 
     d = _symbol_dict(deepcopy(data))
     mt = pop!(d, :model_type, :dtcc)
-    Symbol(mt) == :dtcc || error("Only support dtcc models now")
-
+    Symbol(mt) in (:dtcc, :dtmscc, :dtcscc)  || error("Only support dtcc, dtmscc and dtcscc models now.")
+    if Symbol(mt) == :dtmscc
+      d[:symbols][:exogenous] = pop!(d[:symbols],:markov_states,nothing)
+      d[:options][:exogenous] = pop!(d[:options],:discrete_transition,nothing)
+    elseif Symbol(mt) == :dtcscc
+      d[:symbols][:exogenous] = pop!(d[:symbols],:shocks,nothing)
+      d[:options][:exogenous] = pop!(d[:options],:distribution,nothing)
+    end
     recipe = RECIPES[:dtcc]
     nm = pop!(d, :name, "modeldoesnotwork")
     id = gensym(nm)
     options = pop!(d, :options, Dict{Symbol,Any}())
     defs = pop!(d, :definitions, Dict{Symbol,Any}())
-    exog = pop!(d, :exogenous, Dict{Symbol,Any}())
+    # exog = pop!(d, :exogenous, Dict{Symbol,Any}())
+    exog = get(options, :exogenous, Dict{Symbol,Any}())
```

Comment:
I still don't think we are ensuring that when the model is not dtmscc or dtcscc (i.e. we have a dtcc) that we have exogenous symbols and options.